### ISSUE TITLE
unicode problem with python2.7, lang="de"

### ISSUE THREAD

**Issue by ronen** [savoirfairelinux/num2words#70]
Hello,  thanks for this module!

I came across a unicode problem.  You can see this works with python3:

```bash
$ python3 -c "import num2words; print(num2words.num2words(0.5, lang='de'))"
null Komma fünf null
```

but fails with python 2.7:

```bash
$ python2.7 -c "import num2words; print(num2words.num2words(0.5, lang='de'))"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/local/lib/python2.7/site-packages/num2words/__init__.py", line 50, in num2words
    return converter.to_cardinal(number)
  File "/usr/local/lib/python2.7/site-packages/num2words/base.py", line 93, in to_cardinal
    return self.to_cardinal_float(value)
  File "/usr/local/lib/python2.7/site-packages/num2words/base.py", line 127, in to_cardinal_float
    out.append(str(self.to_cardinal(curr)))
UnicodeEncodeError: 'ascii' codec can't encode character u'\xfc' in position 1: ordinal not in range(128)
```

The problem being the call to `str()` which in python3 returns a unicode string, but in 2.7 `str()` can only return ASCII bytes.  (In 2.7 `unicode()` would need to be called instead of `str()` to get a unicode string)

**Comment by erozqba** [comment]
Hello @ronen thanks for reporting the bug, could you submit a patch and a test?

**Comment by ronen** [comment]
My bad.  This was fixed in 266fb9d2dfa2ea3e94c86c4cf7dd3f88d7bb9fcb and later commits; the machine in question must have been running an older release of num2words.  Closing this out.
